src/neural/loader.cc

```diff
@@ -91,12 +160,15 @@ void PopulateConvBlockWeights(FloatVectors* vecs, Weights::ConvBlock* block) {
 }  // namespace
 
 Weights LoadWeightsFromFile(const std::string& filename) {
-  FloatVectors vecs = LoadFloatsFromFile(filename);
+  FloatVectors vecs;
+  auto buffer = DecompressGzip(filename);
```

Comment:
Check for file size of at least 2.

Reply:
Where exactly? Check if the buffer is >= 2? Or the gzip file itself?

Reply:
I think you can just check the buffer size.  If the file itself is less than 2 bytes, gzopen is going to just assume its not compressed and give you a <2 byte buffer.

Reply:
Indeed, checking buffer size is what's needed.